src/LayerPlan.cpp

```diff
@@ -710,9 +692,67 @@ void LayerPlan::addExtrusionMoveWithGradualOverhang(
         else
         {
             // We cross no border, which means we can reach the end of the segment within the current speed region, so we are done
-            add_extrusion_move(p, actual_speed_region_index);
-            return;
+            extrusion_moves.push_back(SegmentExtrusionMove{ p.toPoint2LL(), actual_speed_region_index });
+            break;
+        }
+    }
+
+    // Filter out micro-segments
+    std::vector<SegmentExtrusionMove> extrusion_moves_filtered;
+    extrusion_moves_filtered.reserve(extrusion_moves.size());
+    Point2LL current_position = start;
+    for (const SegmentExtrusionMove& extrusion_move : extrusion_moves | ranges::views::drop_last(1))
```

Comment:
I think I'd prefer it if we always keep the end-points -- (the end-point is already kept, but) so in the specific case the micro-segment happens between the first and the next, we throw away the next instead of the first.

I say this because we might treat points within 5 units as the same point, bit it can still visually look bad if the corners don't align.

All we'd have to do is also drop the first (or two, depending on the outcome) from the loop and make sure to prepend it separately.

Given we're in a bit of a time-crunch I'll try to add this myself rather than send it back, then send it on to QA.